debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/transforms/outbox/MongoEventRouter.java

```diff
@@ -0,0 +1,301 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.mongodb.transforms.outbox;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.connect.connector.ConnectRecord;
+import org.apache.kafka.connect.data.Field;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.transforms.ExtractField;
+import org.apache.kafka.connect.transforms.Transformation;
+import org.bson.BsonDocument;
+import org.bson.BsonValue;
+import org.bson.json.JsonMode;
+import org.bson.json.JsonWriterSettings;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.config.Configuration;
+import io.debezium.connector.mongodb.transforms.ExtractNewDocumentState;
+import io.debezium.connector.mongodb.transforms.MongoDataConverter;
+import io.debezium.data.Envelope;
+import io.debezium.time.Timestamp;
+import io.debezium.transforms.outbox.EventRouter;
+import io.debezium.transforms.outbox.EventRouterConfigDefinition;
+
+/**
+ * Debezium MongoDB Outbox Event Router SMT
+ *
+ * @author Sungho Hwang
+ */
+public class MongoEventRouter<R extends ConnectRecord<R>> implements Transformation<R> {
+
+    private static final Logger logger = LoggerFactory.getLogger(MongoEventRouter.class);
+
+    private final JsonWriterSettings jsonWriterSettings = JsonWriterSettings.builder()
+            .outputMode(JsonMode.EXTENDED)
+            .indent(true)
+            .newLineCharacters("\n")
+            .build();
+
+    private final MongoDataConverter converter = new MongoDataConverter(ExtractNewDocumentState.ArrayEncoding.ARRAY);
+
+    private String fieldTimestamp;
+    private String fieldPayload;
+    private boolean expandPayload;
+
+    private final ExtractField<R> afterExtractor = new ExtractField.Value<>();
+    private final EventRouter<R> sqlOutboxEventRouter = new EventRouter<>();
+
+    @Override
+    public R apply(R r) {
+        R expandedAfterFieldRecord = r;
+        try {
+            expandedAfterFieldRecord = expandAfterField(r);
+        }
+        catch (Exception e) {
+            // when the record includes update event or
+            // 'after' field is not valid MongoDB data format for converting to Kafka Connect data format
+            logger.warn("Filed to expand after field: " + e.getMessage(), e);
```

Comment:
```suggestion
            logger.warn("Failed to expand after field: " + e.getMessage(), e);
```

Reply:
Can we avoid that exception to begin with, i.e. check whether the expected field is present or not? This is an expected situation (heartbeat events, TX markers, etc.), so this should not trigger an exception, also logs will be cluttered with the warnings superfluously.